common/src/main/java/com/wynntils/utils/UnsignedByteUtils.java

```diff
@@ -0,0 +1,79 @@
+/*
+ * Copyright © Wynntils 2023.
+ * This file is released under LGPLv3. See LICENSE for full license details.
+ */
+package com.wynntils.utils;
+
+import com.wynntils.utils.type.UnsignedByte;
+import java.nio.charset.StandardCharsets;
+
+// FIXME: Unit testing
+public final class UnsignedByteUtils {
+    public static UnsignedByte[] fromBitArray(boolean[] values) {
+        assert values.length % 8 == 0;
+
+        UnsignedByte[] bytes = new UnsignedByte[values.length / 8];
+        for (int i = 0; i < values.length; i += 8) {
+            byte value = 0;
+            for (int j = 0; j < 8; j++) {
+                value |= (values[i + j] ? 1 : 0) << (7 - j);
+            }
+            bytes[i / 8] = UnsignedByte.of(value);
+        }
+        return bytes;
+    }
+
+    public static UnsignedByte[] encodeString(String string) {
+        // Strings are encoded by encoding the char's ASCII value
+        // and is terminated by a 0 byte
+        UnsignedByte[] bytes = new UnsignedByte[string.length() + 1];
+        byte[] asciiBytes = string.getBytes(StandardCharsets.US_ASCII);
```

Comment:
What will happen if the string contains a character that is not representable in US ASCII? Do we get a runtime exception?

Reply:
I'll make it check and throw a readable exception. It'll be passed up to the transformer and nicely displayed to the user.